graylog2-server/src/main/java/org/graylog2/shared/journal/KafkaJournal.java

```diff
@@ -320,52 +309,19 @@ public int getPurgedSegmentsInLastRetention() {
     }
 
     private void setupKafkaLogMetrics(final MetricRegistry metricRegistry) {
-        metricRegistry.register(name(KafkaJournal.class, "size"), new Gauge<Long>() {
-            @Override
-            public Long getValue() {
-                return kafkaLog.size();
-            }
-        });
-        metricRegistry.register(name(KafkaJournal.class, "logEndOffset"), new Gauge<Long>() {
-            @Override
-            public Long getValue() {
-                return kafkaLog.logEndOffset();
-            }
-        });
-        metricRegistry.register(name(KafkaJournal.class, "numberOfSegments"), new Gauge<Integer>() {
-            @Override
-            public Integer getValue() {
-                return kafkaLog.numberOfSegments();
+        metricRegistry.register(name(KafkaJournal.class, "size"), (Gauge<Long>) kafkaLog::size);
+        metricRegistry.register(name(KafkaJournal.class, "logEndOffset"), (Gauge<Long>) kafkaLog::logEndOffset);
+        metricRegistry.register(name(KafkaJournal.class, "numberOfSegments"), (Gauge<Integer>) kafkaLog::numberOfSegments);
+        metricRegistry.register(name(KafkaJournal.class, "unflushedMessages"), (Gauge<Long>) kafkaLog::unflushedMessages);
+        metricRegistry.register(name(KafkaJournal.class, "recoveryPoint"), (Gauge<Long>) kafkaLog::recoveryPoint);
+        metricRegistry.register(name(KafkaJournal.class, "lastFlushTime"), (Gauge<Long>) kafkaLog::lastFlushTime);
+        metricRegistry.register(GlobalMetricNames.JOURNAL_OLDEST_SEGMENT, (Gauge<Date>) () -> {
```

Comment:
This works when using an anonymous class. Apparently the Jackson serialization doesn't properly go through lambdas here.